coops-ui/src/views/AnnualReport.vue

```diff
@@ -161,19 +162,78 @@ export default {
     },
 
     submit () {
-      // TODO - redirect to payment - will need to save state of page, etc
-      // TODO - make proper axios call and delete hardcoded data below
-      // TODO - other team doing credit card entering/payment confirmation? don't know what to check for in result
-      // const url = this.payURL
-      // axios.get(url)
-      //   .then(response => {
-      //     if (response && response.data) {
-      //       this.setFiledDate(this.currentDate)
-      //     } else {
-      //       console.log('submit() error - invalid response data')
-      //     }
-      //   })
-      //   .catch(error => console.error('submit() error =', error))
+      let changeOfDirectors = null
+      let changeOfAddress = null
+
+      const header = {
+        header: { name: 'annual_report', date: this.currentDate }
+      }
+      const business = {
+        business: {
+          foundingDate: this.entityFoundingDate,
+          identifier: this.entityIncNo,
+          legalName: this.entityName
+        }
+      }
+
+      const annualReport = {
+        annualReport: {
+          annualGeneralMeetingDate: this.agmDate,
+          certifiedBy: 'full name',
+          email: 'no_one@never.get'
+        }
+      }
+
+      if (this.isDataChanged('OTCDR')) {
+        changeOfDirectors = {
+          changeOfDirectors: {
+            certifiedBy: 'Full Name',
+            email: 'no_one@never.get',
+            directors: this.$refs.directors.getAllDirectors()
```

Comment:
$refs.directors gets a js error - there is no ref called "directors". I understand that you are referring to the Directors component above, and I tried out a fix with ref="directors" and it also doesn't work, probably because it's a conflict with the name of the component. If you add ref="directorsList" and use $refs.directorsList here, that will work.